src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3622,16 +3622,23 @@ def _build_identities_info_from_system_user_assigned(cmd, mi_system_assigned, mi
     return identity_types, user_assigned_identities
 
 
-def deallocate_vmss(cmd, resource_group_name, vm_scale_set_name, instance_ids=None, no_wait=False):
+def deallocate_vmss(cmd, resource_group_name, vm_scale_set_name, instance_ids=None, no_wait=False, hibernate=None):
     client = _compute_client_factory(cmd.cli_ctx)
-    if instance_ids and len(instance_ids) == 1:
+    # a walkaround because `VirtualMachineScaleSetVMsOperations.begin_deallocate` does not accept `hibernate`
+    # https://azuresdkdocs.blob.core.windows.net/$web/python/azure-mgmt-compute/30.0.0/azure.mgmt.compute.v2023_03_01
+    # .operations.html#azure.mgmt.compute.v2023_03_01.operations.VirtualMachineScaleSetVMsOperations
+    if instance_ids and len(instance_ids) == 1 and hibernate is None:
```

Comment:
Please confirm if the REST supports `hibernate`. If so, we should consider ask service team to upgrade their SDK to support `hibernate` instead of adding special logic for the missing content in the SDK, which will also introduce unnecessary request calls